config/application.rb

```diff
@@ -76,16 +76,24 @@ class Application < Rails::Application
     config.external_url_mapping = APP_CONFIG = YAML.load(File.read(Rails.root + "config/external_link_mapping.yml"))
 
     config.before_initialize do
+      Settings.prepend_source!(Rails.root.join('config', 'hets.yml').to_s)
+      Settings.reload!
+
       # Enable serving of images, stylesheets, and JavaScripts from an asset server
       config.action_controller.asset_host = Settings.asset_host
 
       # ActionMailer settings
-      c = Settings.action_mailer
-      (c[:default_url_options] ||= {})[:host] ||= Settings.hostname
-      c.each do |key,val|
-        config.action_mailer.send("#{key}=", val)
+      require config.root.join('config', 'initializers', 'hostname.rb')
```

Comment:
Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping.